libs/components/popovers/src/lib/modules/popover/popover.directive.ts

```diff
@@ -14,12 +21,36 @@ import { SkyPopoverTrigger } from './types/popover-trigger';
   selector: '[skyPopover]',
 })
 export class SkyPopoverDirective implements OnInit, OnDestroy {
+  /**
+   * Appends the `sky-popover-trigger` class to the trigger element.
+   * @internal
+   */
+  @HostBinding('class')
+  protected directiveClass = 'sky-popover-trigger';
+
+  /**
+   * Appends the `data-popover-id` attribute to the trigger element set to a unique ID generated by the popover component.
+   * @internal
+   */
+  // TODO: replace this with relevant ARIA attributes
+  @HostBinding('[attr.data-popover-id]')
+  protected popoverId = 'unknown';
```

Comment:
Any way we can make this `string | undefined`? "Undefined" programmatically communicates an unset value better than `'unknown'`.

Reply:
Seems to be fine. Changed